src/components/TreeGrid.ts

```diff
@@ -46,31 +48,30 @@ const wrapContent =
     state: 'closed',
   });
 
-export const hierarchyMapToRows = <T extends RowContents>(hierarchyMap: Map<T, T[]>): Row<T>[] => {
-  const traverseHierarchy = (parent: T, depth: number, previousRows: Row<T>[]): Row<T>[] => {
+export const populateTreeFromRoot = <T extends RowContents>(root: T): Row<T>[] => {
+  const createRows = (parent: RowContents, depth: number, previousRows: Row<T>[]): Row<T>[] => {
     // does parent have children?
-    const children = hierarchyMap.get(parent) || [];
+    const children = parent.children ?? [];
     const parentRow: Row<T> = {
-      contents: parent,
+      contents: parent as T,
       depth,
       isFetched: children.length > 0,
       state: children.length > 0 ? 'open' : 'closed',
     };
-    // recursively traverse hierarchy of all children
-    const childRows = children.flatMap((child) => traverseHierarchy(child, depth + 1, []));
+    // For each child, generate its rows. Add the generated rows to the end.
+    const childRows = children.flatMap((child) => createRows(child, depth + 1, []));
     return [...previousRows, parentRow, ...childRows];
   };
 
-  // hierarchyMap assumes that the root is the first map entry
-  // using depth of -1 because we don't want to show the root
-  return _.tail(traverseHierarchy(hierarchyMap.keys().next().value, -1, []));
+  // To hide the root, call tail() and use a depth of -1, so the root's children are at depth 0.
+  return _.tail(createRows(root, -1, []));
 };
 
 type TreeGridProps<T extends RowContents> = {
   /** the columns to display */
   readonly columns: Column<T>[];
-  /** the initial rows to display */
-  readonly initialHierarchy: Map<T, T[]>;
+  /** The root of the tree to display. Note that the root itself is hidden. */
```

Comment:
I don't really understand the meaning of this comment. Could it be clarified/elaborated?